### PR TITLE
Move container instance health doctor to ecs-agent/

### PR DESCRIPTION
**Pull request by danehlim**
### Summary

Move doctor (container instance health) to `ecs-agent/` and reformat accordingly.

NOTE: An interface for logging to be used in `ecs-agent/` will be added as part of a future PR and is out of scope of this PR. For now, we continue to leverage our existing dependency on seelog in the ecs-agent module.

### Implementation details

- Move `agent/doctor/doctor.go`, `agent/doctor/doctor_test.go`, `agent/doctor/healthcheck.go`, `agent/doctor/healthcheckstatus.go`, and `agent/doctor/healthcheckstatus_test.go` to `ecs-agent/doctor/` directory
- Replace all imports of `github.com/aws/amazon-ecs-agent/agent/doctor` with `github.com/aws/amazon-ecs-agent/ecs-agent/doctor` and then resort imports
- For files that previously imported `github.com/aws/amazon-ecs-agent/agent/doctor` but still have a dependency on `agent/doctor/docker_runtime_healthcheck.go`, reimport `github.com/aws/amazon-ecs-agent/agent/doctor` with alias `dockerdoctor` (to avoid package name conflict with `github.com/aws/amazon-ecs-agent/ecs-agent/doctor`)
   - `github.com/aws/amazon-ecs-agent/ecs-agent/doctor` is for general container instance health check functionality and constants
   - `github.com/aws/amazon-ecs-agent/agent/doctor` is for container instance health check things that are specific to our Agent, such as Docker health checks (since our Agent is Docker based)
- Update doctor tests to use correct expected and actual values for assertions
- Update health check status tests to use assert package for assertions
- Go into `agent/`  directory and run `go mod tidy` and `go mod vendor`
- Go into `ecs-agent/` directory and run `go mod tidy` and `go mod vendor`

### Testing

Unit, integration, and functional tests.

### Description for the changelog

Move container instance health doctor to ecs-agent/

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/go.mod [hunk 1]
```diff
@@ -3,23 +3,41 @@ module github.com/aws/amazon-ecs-agent/ecs-agent
...
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	golang.org/x/mod v0.8.0 // indirect
 	golang.org/x/net v0.8.0 // indirect
 	golang.org/x/sys v0.6.0 // indirect
+	golang.org/x/text v0.8.0 // indirect
 	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/aws/amazon-ecs-agent/agent => ../agent
```
Comment (amogh09) [inline ecs-agent/go.mod:43]: Why do we need this dependency? `ecs-agent` depends on `agent`, having the reverse dependency seems odd.

Comment (danehlim) [reply ecs-agent/go.mod]: Discussed offline, aligned that we should continue to leverage our existing dependency on seelog in the ecs-agent module until a logging interface to be used in ecs-agent module has been added in a future PR. I have updated this PR's changes and description accordingly. Thanks again for calling out!

Comment (danehlim) [reply ecs-agent/go.mod]: Discussed offline, aligned that we should continue to leverage our existing dependency on seelog in the ecs-agent module until a logging interface to be used in ecs-agent module has been added in a future PR. I have rebased and updated this PR's changes and description accordingly. Thanks again for calling out!

File: ecs-agent/doctor/doctor_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	assert.Len(t, newDoctor.healthchecks, 0)
+	newTestHealthcheck := &trueHealthcheck{}
+	newDoctor.AddHealthcheck(newTestHealthcheck)
+	assert.Len(t, newDoctor.healthchecks, 1)
+}
+
+func TestGetCluster(t *testing.T) {
+	clusterName := "test-cluster"
+	newDoctor := Doctor{cluster: clusterName}
+	assert.Equal(t, newDoctor.GetCluster(), clusterName)
```
Comment (amogh09) [inline ecs-agent/doctor/doctor_test.go:92]: nit: `assert.Equal` function takes the expected value before the actual value. 

[func Equal(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) bool](https://pkg.go.dev/github.com/stretchr/testify@v1.8.2/assert#Equal)

Not too big a deal but if the test fails then the values against "Expected" and "Actual" will be printed incorrectly.

File: ecs-agent/doctor/healthcheckstatus_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+type testHealthcheckStatus struct {
+	SomeStatus HealthcheckStatus `json:"status"`
+}
+
+func TestUnmarshalHealthcheckStatus(t *testing.T) {
+	status := HealthcheckStatusInitializing
+
+	err := json.Unmarshal([]byte(`"INITIALIZING"`), &status)
+	if err != nil {
+		t.Error(err)
```
Comment (amogh09) [inline ecs-agent/doctor/healthcheckstatus_test.go:44]: nit: we could use the assert package for the assertions.

**Review by amogh09 — APPROVED** [review]
LGTM overall. Left a couple of minor comments.

File: ecs-agent/doctor/healthcheckstatus_test.go (hunk 3 above)
Comment (danehlim) [reply ecs-agent/doctor/healthcheckstatus_test.go:46]: Updated to use assert package.

File: agent/app/agent.go [hunk 4]
```diff
@@ -23,8 +23,9 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/logger"
 	"github.com/aws/amazon-ecs-agent/agent/logger/field"
 
-	"github.com/aws/amazon-ecs-agent/agent/doctor"
+	dockerdoctor "github.com/aws/amazon-ecs-agent/agent/doctor"
```
Comment (YashdalfTheGray) [inline agent/app/agent.go]: The distinction between `dockerdoctor` and `doctor` might not be clear to someone who comes upon this code, can we just throw a comment in here?

File: agent/doctor/docker_runtime_healthcheck_test.go [hunk 5]
```diff
@@ -16,33 +17,33 @@ func TestNewDockerRuntimeHealthCheck(t *testing.T) {
...
+	assert.Equal(t, doctor.HealthcheckStatusInitializing, dockerRuntimeHealthCheck.Status)
 }
 
 func TestRunCheck(t *testing.T) {
 	testcases := []struct {
 		name               string
 		dockerPingResponse *dockerapi.PingResponse
-		expectedStatus     HealthcheckStatus
-		expectedLastStatus HealthcheckStatus
+		expectedStatus     doctor.HealthcheckStatus
```
Comment (YashdalfTheGray) [inline agent/doctor/docker_runtime_healthcheck_test.go:27]: It actually might be worthwhile to include (potentially in the PR description's point of the name of the import changing) that the `doctor` import is the interface and the constants, while `dockerdoctor` is the implementation bits that are specific to the Docker agent. We'll likely run into this more as we pull more of the interfaces out and leave the implementations in place.

**Review by YashdalfTheGray — APPROVED** [review]

**Review by amogh09 — DISMISSED** [review]

File: agent/app/agent.go (hunk 4 above)
Comment (danehlim) [reply agent/app/agent.go]: Good idea, will do.

**Review by amogh09 — APPROVED** [review]

**Review by YashdalfTheGray — APPROVED** [review]
Code looks good, thanks for making the updates.

**Pull request merged**